sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -316,16 +316,38 @@ static const struct snd_kcontrol_new controls[] = {
 	SOC_DAPM_PIN_SWITCH("Speaker"),
 };
 
+static int set_amp_tdm_slots(struct snd_soc_pcm_runtime *rtd, bool tdm_mode)
+{
+	struct snd_soc_dai *codec_dai;
+	int ret;
+	int i;
+
+	for_each_rtd_codec_dai(rtd, i, codec_dai) {
+		if (tdm_mode)
+			/* tx_mask 0, rx_mask with only one bit, single slot) */
+			ret = snd_soc_dai_set_tdm_slot(codec_dai, 0, BIT(i), 1, 0);
```

Comment:
@plbossart   with the help of @keqiao,  I finally understand why we can't set LL to both 1308. 
1308-1  ch_mask = 0x1,  num = 1  // only ch0 is enabled in 1308-1
1308-2 ch_mask = 0x2, num = 1   // only ch1 is enabled in 1308-2 and ch0 is disabled,  so LL can't work for this slave.

we need set ch_mask = 1 to all single channel slave to enable ch0 so that LL can work. I confirmed this on TGL.
I almost forgot my change, then when I checked channel-enable register in spec, I suddenly recalled everything.

Reply:
what do you mean by 'set ch_mask = 1'?
Where would that be done? in the rt1308-sdw.c code?

Reply:
sorry,  I mean : " snd_soc_dai_set_tdm_slot(codec_dai, 0, 1, 1, 0);"   here rx_mask should be set to 1, so that ch_mask = 1 in rt1308-sdw.c code

Reply:
@RanderWang it makes no sense to use the set_tdm slot to assign the same slot to everyone. It's really not what the API is about.

I think the RT1308 driver has a different understanding of what the ch_mask is. For SoundWire it means 'which of the channels should be enabled' for that data port.

Reply:
Or how about do it in sdw-1308.c:
ch_mask = BIT(hweight_long(rt1308->rx_mask)) - 1;  \\\height_long counts number of bit 1
rx_mask = 0x0,  ch_mask = 0x0,
rx_mask = 0x1,  ch_mask = 0x1,
rx_mask = 0x2,  ch_mask = 0x1,
rx_mask = 0x3,  ch_mask = 0x3,
rx_mask = 0x4,  ch_mask = 0x1,
rx_mask = 0x5,  ch_mask = 0x3,

Reply:
I don't think that would work @RanderWang. The ch_mask is linked to the block offset, if you modify it then we likely read the wrong data from the bus, e.g. with rx_mask = 0x2 we want to read the ch1. with the formula you suggest we'd read ch0.

Reply:
@plbossart  It works, I confirmed it on TGL.  please check the following picture, the key idea is: the ch1 data is the input of left channel for 1308-2, not right channel.

````
----------
|  ch0   |        ---->  1308-1 (left channel)
----------
|  ch1   |        ---->  1308-2 (left channel)
----------
````

Reply:
please forget 1308 is a two-channel codec, now it is set to one-channel codec, so he only care about left channel or the first channel.